src/Npgsql/NpgsqlConnection.cs

```diff
@@ -1228,6 +1228,9 @@ public bool Wait(int timeout)
         [PublicAPI]
         public Task WaitAsync(CancellationToken cancellationToken)
         {
+            if (cancellationToken.IsCancellationRequested)
+                return PGUtil.CancelledTask;
```

Comment:
Should `Task.FromCanceled(cancellationToken)` be used here?

Reply:
Unfortunately it doesn't exist in net45 which is targeted in Npgsql 4.0... The equivalent for dev is using `Task.FromCanceled()` (#2443).